http/router/matching.lua

```diff
@@ -70,13 +70,20 @@ local function matches(r, filter)
     end
 
     local regex_groups_matched = {string.match(filter.path, r.match)}
+
     if #regex_groups_matched == 0 then
         return false
     end
-    if #r.stash > 0 and #r.stash ~= #regex_groups_matched then
+    if #r.stash > 0 and #r.stash ~= #regex_groups_matched - 1 then
         return false
     end
 
+    -- first regex_group is whole path, we should omit it
```

Comment:
Why not simply `table.remove(regex_groups_matched, 1)`?

Reply:
Yes it's simpler and clearer.